api/lib/mno_enterprise/concerns/controllers/jpi/v1/user_access_requests_controller.rb

```diff
@@ -15,35 +15,70 @@ def index
     @user_access_requests = MnoEnterprise::UserAccessRequest.active_requested(current_user.id)
   end
 
+  # GET /mnoe/jpi/v1/last_access_request
+  def last_access_request
+    @user_access_request = MnoEnterprise::UserAccessRequest.last_access_request(current_user.id)
+    render 'show'
+  end
+
+  # POST /mnoe/jpi/v1/user_access_requests
+  def create
+    @user_access_request = MnoEnterprise::UserAccessRequest.new(create_params)
+    @user_access_request.relationships.user = current_user
+    @user_access_request.save
+    MnoEnterprise::EventLogger.info('access_authorized', current_user.id, 'Authorize Access', user_access_request)
+    @user_access_request = @user_access_request.load_required(:requester)
+    render 'show'
+  end
+
   # PUT /mnoe/jpi/v1/user_access_requests/:id/deny
   def deny
     return render_not_found('user_access_request') unless user_access_request
-    @user_access_request = user_access_request.deny.first
-    if @user_access_request.errors.empty?
+    result = user_access_request.deny
+    if result.errors.empty?
       @user_access_request = @user_access_request.load_required(:requester)
       MnoEnterprise::EventLogger.info('access_denied', current_user.id, 'Access denied', @user_access_request)
       MnoEnterprise::SystemNotificationMailer.access_denied(user_access_request.id).deliver_later
       render 'show'
     else
-      render_bad_request('deny', @user_access_request.errors)
+      render_bad_request('deny', result.errors.full_messages)
     end
   end
 
   # PUT /mnoe/jpi/v1/user_access_requests/:id/approve
   def approve
     return render_not_found('user_access_request') unless user_access_request
-    @user_access_request = user_access_request.approve.first
-    if @user_access_request.errors.empty?
-      @user_access_request = @user_access_request.load_required(:requester)
+    result = user_access_request.approve!()
+    if result.errors.empty?
       MnoEnterprise::EventLogger.info('access_approved', current_user.id, 'Access approved', @user_access_request)
       MnoEnterprise::SystemNotificationMailer.access_approved(user_access_request.id).deliver_later
       render 'show'
     else
-      render_bad_request('approve', @user_access_request.errors)
+      render_bad_request('approve', result.errors.full_messages)
+    end
+  end
+
+  # PUT /mnoe/jpi/v1/user_access_requests/:id/revoke
+  def revoke
+    return render_not_found('user_access_request') unless user_access_request
+    result = user_access_request.revoke
+    if result.errors.empty?
+      @user_access_request = @user_access_request.load_required(:requester)
+      MnoEnterprise::EventLogger.info('access_revoked', current_user.id, 'Access revoked', @user_access_request)
+      render 'show'
+    else
+      render_bad_request('revoke', result.errors.full_messages)
     end
   end
 
   def user_access_request
     @user_access_request ||= MnoEnterprise::UserAccessRequest.find_one(params[:id])
   end
+
+  def create_params
```

Comment:
All good but put a comment here or on the create to explain what is happening: the user is pre-approving access to his account as this is not super intuitive.